### PR TITLE
Implement the physique of the minecraft swim

### LINKED ISSUE #25: Swimming animation two blocks under water block
https://github.com/NukkitX/Nukkit/issues/785

### PR DESCRIPTION
**Pull request by tuilakhanh**
Fix #25 

Todo 
- [x]  Do not take damage while inside the 1 block height hole

### CODE REVIEW

**Comment by tuilakhanh** [comment]
Ready to review

**Review by joserobjr — CHANGES_REQUESTED** [review]
Thank you for your PR! I got your idea but I don't think it's the best way to do it. I'm going to do some changes before merging.

**Review by joserobjr — APPROVED** [review]
All good for me now :)

File: src/main/java/cn/nukkit/entity/EntityHuman.java [hunk 1]
```diff
@@ -50,14 +52,21 @@ public float getHeight() {
...
     public float getEyeHeight() {
-        return 1.62f;
+        return (float)(boundingBox.getMaxY() - boundingBox.getMinY() - 0.18);
     }
 
     @Override
     protected float getBaseOffset() {
-        return this.getEyeHeight();
+        return 1.62f;
     }
```
Comment (roridev) [inline src/main/java/cn/nukkit/entity/EntityHuman.java:71]: On this change i'm confused. why did getBaseOffset got inlined to `1.62f` when previously it was the same as `getEyeHeight`, while getEyeHeight became a complex calculation.

From this i can spot a problem that could arise.
When a player is swimming and a plugin requests `getBaseOffset` the offset given to it could desync with the actual player height (since its a set inline value, and the bouding box now changes to approx. a meter cubed and 1.62 being +.62 above the aabb). Was this case considered?

Comment (joserobjr) [reply src/main/java/cn/nukkit/entity/EntityHuman.java:71]: The `BaseOffset` is the height difference between the packet communication and the actual position on the server. It usually doesn't change and before we could swim, it was the same as the eye height for the human entities, so somebody might have used getEyeHeight as a shortcut.

The `EyeHeight` is where the eyes are located in a height considering that the player is on y 0. When the player starts swimming, the player bounding box reduces in height, so does the eye height. Since this is not a constant value (thank god we have getters here), the value is adjusted depending on the player's current height. I don't know if sneaking also changes the player's height in Bedrock Edition, if it does, then we can also benefit from this change to return the right eye height.

Long story short, both used to return the same  constant values, now only `getBaseOffset()` return the old value and `getEyeHeight()` returns the current player eye height.

Comment (roridev) [reply src/main/java/cn/nukkit/entity/EntityHuman.java:71]: Understood.

**Review by roridev — APPROVED** [review]

**Comment by joserobjr** [comment]
Thank you all xD
